### ISSUE TITLE
YOLOv11-Seg : How to make input data from torch.Tensor or numpy.array, not from yaml

### ISSUE THREAD

**Issue by KatnissSnow** [ultralytics/ultralytics#18158]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I would like to know if it is possible to train YOLOv11-Seg with a dataloader whose images are generated before training but not stored. input data is the result from other deepnet, so the data is torch.Tensor or numpy.array. how i change the code(model.train(data="coco8-seg.yaml", epochs=100, imgsz=640)) to make input data not from yaml.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @KatnissSnow, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples, which may provide insights for your requirements.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example (MRE)](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us debug it.

If this is a custom training ❓ Question, as indicated, could you please clarify further or provide additional context to your request? The method you are describing (inputting data from a `torch.Tensor` or `numpy.array` rather than a YAML file) may involve customizing the data loading pipeline. Providing more details such as the structure of your data, your workflow goals, or sample training logs will help us guide you better. Additionally, make sure to review our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best:
- For real-time discussions, connect with us on [Discord](https://discord.com/invite/ultralytics) 🎧
- Engage in detailed discussions on our [Discourse](https://community.ultralytics.com)
- Share knowledge and ideas with others on our [Subreddit](https://reddit.com/r/Ultralytics)

## Upgrade

Ensure you're using the latest version of the `ultralytics` package. You can upgrade by running the following command in a clean [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

You can see all requirements [here](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml).

## Environments

YOLO works seamlessly in the following verified environments (with all dependencies preinstalled):
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are currently passing. These tests ensure that all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) work correctly across macOS, Windows, and Ubuntu systems. 

This is an automated response to help address your query faster, but rest assured that an Ultralytics engineer will also assist you soon 😊

**Comment by Y-T-G** [comment]
Custom dataloader is not supported.
